perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/MuadmPasswordManagerModuleTest.java

```diff
@@ -0,0 +1,62 @@
+package cz.metacentrum.perun.core.impl.modules.pwdmgr;
+
+import cz.metacentrum.perun.core.AbstractPerunIntegrationTest;
+import cz.metacentrum.perun.core.api.exceptions.PasswordStrengthException;
+import cz.metacentrum.perun.core.bl.PerunBl;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+public class MuadmPasswordManagerModuleTest extends AbstractPerunIntegrationTest {
+
+	private MuadmPasswordManagerModule module;
+
+	@Before
+	public void setUp() {
+		this.module = (MuadmPasswordManagerModule) ((PerunBl) sess.getPerun()).getUsersManagerBl()
+			.getPasswordManagerModule(sess, "mu-adm");
+	}
+
+	@Test
+	public void weakPasswordsTest() {
+		List<String> weakPasswords = List.of(
+			"eakfjklsjweiml",
+			"LOSFNXHRKJANCS",
+			"18746982365414",
+			"!@#$%^&*_-+=?/",
+			"AkoHCNqrXMsuqm",
+			"afgr487xiy2m63",
+			"!lop@#twoep&*/",
+			"RT672HWZ1068KK",
+			"A!@#KDSAJ?/NMO",
```

Comment:
This test definitely covers a lot of cases. However, it is a good practice to test only a single case, which might fail, in a single test. It is also a good practice to put the name of the tested case to the test name e.g.: `loginInPasswordFails` or `reverseLoginInPasswordFails`. 

This is because it is much clearer, what the purpose of a single test is. Also, when some of these tests fail, you immediately know what has failed. With this kind of test, you would need to debug it first to see what failed.